app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, only: %i[new create destroy edit]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all.order('created_at DESC')
+    @post = Post.new
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
```

Comment:
- **_The new action should be initializes as an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**